### PR TITLE
Don't impersonate when trying to write to system metadata store

### PR DESCRIPTION
**Pull request by gokulavasan**

### CODE REVIEW

**Comment by anwar6953** [comment]
LGTM

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java [hunk 1]
```diff
@@ -104,20 +104,27 @@ public DatasetSpecification createOrUpdate(final Id.DatasetInstance datasetInsta
...
             throw new BadRequestException(
               String.format("Cannot instantiate dataset type using provided type meta: %s", typeMeta));
           }
-          DatasetSpecification spec = existing == null
-            ? type.configure(datasetInstanceId.getId(), props)
+          return type;
+        }
+      });
+
+      DatasetSpecification spec = ImpersonationUtils.doAs(ugi, new Callable<DatasetSpecification>() {
```
Comment (bdmogal) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java:122]: what's the cost of doing `doAs` twice? Should we combine the two callables, have declarations outside, only initialize them inside the callable, then write system metadata outside the callable after that?

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java:122]: the cost of doing this `doAs` is minimal. It doesn't involve any extra RPC or anything.

Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java:122]: there isn't a way to figure out system namespace in ImpersonationUtils? i know it gets only ugi, just wondering if theres' a way.

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java:122]: Not necessary to have the check there. It is already provided a fixed UGI.

**Pull request merged**